### ISSUE TITLE
HA color picker not retaining color choice.

### ISSUE THREAD

**Issue by ullms1** [jason0x43/hacs-hubitat#85]
When setting color of my RGB Osram Light in HA using the color picker,  the bulb changes color but the color state attribute doesn't appear.  And when you go back and look at the color picker the picker reverts back to center/white.  Bulb stays the color picked the first time.  log shows HA received the update.

```
2020-11-18 21:31:26 DEBUG (MainThread) [custom_components.hubitat.light] Turning on LIGHT-LVR-BR with {'hs_color': (270.0, 71.42857142857142)}
2020-11-18 21:31:26 DEBUG (MainThread) [hubitatmaker.hub] Sending command on() to 34
2020-11-18 21:31:26 DEBUG (MainThread) [custom_components.hubitat.device] sent on to 34
2020-11-18 21:31:26 DEBUG (MainThread) [hubitatmaker.hub] Sending command setColor({"hue": 75, "saturation": 71.42857142857142}) to 34
2020-11-18 21:31:26 DEBUG (MainThread) [custom_components.hubitat.device] sent setColor to 34
2020-11-18 21:31:28 DEBUG (MainThread) [hubitatmaker.hub] Received event: {'name': 'hue', 'value': '75', 'displayName': 'LIGHT-LVR-BR', 'deviceId': '34', 'descriptionText': 'LIGHT-LVR-BR hue was set to 75%', 'unit': '%', 'type': None, 'data': None}
2020-11-18 21:31:28 DEBUG (MainThread) [hubitatmaker.hub] Updating hue of 34 to 75
2020-11-18 21:31:28 DEBUG (MainThread) [hubitatmaker.hub] Received event: {'name': 'colorName', 'value': 'Violet', 'displayName': 'LIGHT-LVR-BR', 'deviceId': '34', 'descriptionText': 'LIGHT-LVR-BR color is Violet', 'unit': None, 'type': None, 'data': None}
2020-11-18 21:31:28 DEBUG (MainThread) [hubitatmaker.hub] Updating colorName of 34 to Violet
2020-11-18 21:31:54 DEBUG (MainThread) [hubitatmaker.hub] Received event: {'name': 'lastActivity', 'value': '2020 Nov 18 Wed 9:31:51 PM', 'displayName': 'Dimmer-LVR2', 'deviceId': '1036', 'descriptionText': None, 'unit': None, 'type': None, 'data': None}
```

**Comment by jason0x43** [comment]
This sounds like it could be related to #72.

**Comment by jason0x43** [comment]
From the log, it looks like Hubitat isn't reporting the updated saturation. When you change a value through HA, HA sends a command to Hubitat and temporarily updates the color picker UI. If HA doesn't receive confirmation of the change after a short timeout (a couple of seconds), it reverts its internal state to whatever it was before the color was changed. Assuming the original saturation was 0, then that's what it will reset to when Hubitat doesn't send a saturation update event (and a saturation of 0 is white).

Just for comparison, setting the color with a Virtual RGB bulb generates a log like:

```
2020-11-26 10:32:43 DEBUG (MainThread) [custom_components.hubitat.light] Turning on Virtual RGB light with {'hs_color': (270.0, 85.71428571428571)}
2020-11-26 10:32:43 DEBUG (MainThread) [hubitatmaker.hub] Sending command on() to 2178
2020-11-26 10:32:43 DEBUG (MainThread) [custom_components.hubitat.device] sent on to 2178
2020-11-26 10:32:43 DEBUG (MainThread) [hubitatmaker.hub] Sending command setColor({"hue": 75, "saturation": 85.71428571428571}) to 2178
2020-11-26 10:32:43 DEBUG (MainThread) [custom_components.hubitat.device] sent setColor to 2178
2020-11-26 10:32:43 DEBUG (MainThread) [hubitatmaker.hub] Received event: {'name': 'hue', 'value': '75', 'displayName': 'Virtual RGB light', 'deviceId': '2178', 'descriptionText': 'Virtual RGB light hue was set to 75%', 'unit': '%', 'type': None, 'data': None}
2020-11-26 10:32:43 DEBUG (MainThread) [hubitatmaker.hub] Updating hue of 2178 to 75
2020-11-26 10:32:43 DEBUG (MainThread) [hubitatmaker.hub] Received event: {'name': 'saturation', 'value': '85', 'displayName': 'Virtual RGB light', 'deviceId': '2178', 'descriptionText': 'Virtual RGB light saturation was set to 85%', 'unit': '%', 'type': None, 'data': None}
2020-11-26 10:32:43 DEBUG (MainThread) [hubitatmaker.hub] Updating saturation of 2178 to 85
2020-11-26 10:32:43 DEBUG (MainThread) [hubitatmaker.hub] Received event: {'name': 'colorName', 'value': 'Violet', 'displayName': 'Virtual RGB light', 'deviceId': '2178', 'descriptionText': 'Virtual RGB light colorName is Violet', 'unit': None, 'type': None, 'data': None}
2020-11-26 10:32:43 DEBUG (MainThread) [hubitatmaker.hub] Updating colorName of 2178 to Violet
```

What driver are you using for these bulbs in Hubitat?

> the bulb changes color but the color state attribute doesn't appear

What state attribute did you mean?

**Comment by ullms1** [comment]
Sorry I said RGB bulb in my opening post.  I think the issue is with RGBW lights (using Generic Zigbee RGBW Driver with my Osram Lightify downlights). 

RGB lights work fine.

What I meant by attributes was I'm not seeing the color attribute in developer tools states section.

**Comment by ullms1** [comment]
I have tried to replicated it just using the Virtual RGB and Virtual RGBW drivers as well.  The RGB works fine.  RGBW doesn't update the color picker.  And yes this sounds exactly like the issue stated in #72 now that you point it out.

Thanks

**Comment by jason0x43** [comment]
Thanks for looking into that. I can confirm that a light using the Virtual RGBW driver behaves quite differently from a light using the Virtual RGB driver, so that should help track down the issue.

**Comment by jason0x43** [comment]
There was a typo in a constant that caused RGBW lights to not report their color state. This is fixed in  v0.5.12.

**Comment by ullms1** [comment]
Awesome!  Thanks Jason.
